### PR TITLE
Add Breast Cancer Prediction Model

### PR DESCRIPTION
**Pull request by murtaza-sadri-19**
I have a updated the structure using new repository because the commits were conflicting with each other.
@yashasvini121 kindly let me know about any further changes.

### CODE REVIEW

**Comment by murtaza-sadri-19** [comment]
Hi @yashasvini121,
I am sorry that this mess happened but I managed to adhere to Project Structure. Please review this and if any changes please let me know

**Review by yashasvini121 — APPROVED** [review]

**Review by yashasvini121 — COMMENTED** [review]
Everything looks great so far! However, the page for this model hasn't been created yet. Refer the tutorial on how to create the page. Additionally, make sure to update the `app.py` file accordingly to list your model page. You can ignore model_details for now.

**Comment by murtaza-sadri-19** [comment]
Hi @yashasvini121, I have updated the app.py file.

And I'm confused that is the PR merged or approved. Please help me regarding this.

**Comment by yashasvini121** [comment]
@murtaza-sadri-19, your earlier changes were correct, so those changes were been approved, but not merged.

As long as the PR shows as open, it has not been merged.

**Comment by murtaza-sadri-19** [comment]
> @murtaza-sadri-19, your earlier changes were correct, so those changes were approved but not merged.
> 
> As long as the PR shows as open, it has not been merged.
> 
> P.S. The `app.py` and `pages.json` files have not been updated yet, hope you are working on that.

Got it @yashasvini121, Thanks for helping.
And I have updated App.py, pages and form_configs for Breast Cancer Detection.

**Comment by murtaza-sadri-19** [comment]
Hi @yashasvini121, can you review the changes I have made under App.py, pages and form_configs for Breast Cancer Detection.

File: pages/Breast_cancer_Detector.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from page_handler import PageHandler
+
+page_handler = PageHandler("pages/pages.json")
+page_handler.render_page("Breast Tumor Classifier Form")
```
Comment (yashasvini121) [inline pages/Breast_cancer_Detector.py:4]: Page not found, please update `pages/pages.json`.

**Review by yashasvini121 — CHANGES_REQUESTED** [review]
You need to add your page details to `pages/pages.json` for it to be rendered by the `page_handler` class.

File: models/Breast_Cancer_Detector/predict.py
Comment (yashasvini121) [inline models/Breast_Cancer_Detector/predict.py:1]: `predict.py` and `model.py` have similar codes, either update them, i.e. make them depend on one another, or remove the `model.py` file. and update the code accordingly i.e. fix `pages.json`.

If it gets complicated then leave the file as it is.

File: models/Breast_Cancer_Detector/predict.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# models/breast_cancer/model.py
+
+import pickle
+import numpy as np
+
+Loaded_model = pickle.load(open("D:\DS_March\Projects\Breast_cancer_model.pkl", 'rb'))
```
Comment (yashasvini121) [inline models/Breast_Cancer_Detector/predict.py:6]: Write relative file path over here

File: models/Breast_Cancer_Detector/notebooks/Breast_cancer_Prediction_Ensemble_Hyperparameter.ipynb
Comment (yashasvini121) [inline models/Breast_Cancer_Detector/notebooks/Breast_cancer_Prediction_Ensemble_Hyperparameter.ipynb:1]: Add a `model_details()` function over here.
The predict.py file should include a properly implemented model_details() function, or the notebook should contain this function to print a detailed model report. The model will not be accepted without this function, as it is essential for generating the necessary model details.

Sorry if I forgot to mention this earlier. Let me know if you have any confusion regarding the `model_details()`

**Review by yashasvini121 — CHANGES_REQUESTED** [review]

**Comment by murtaza-sadri-19** [comment]
Hi @yashasvini121 , I have modified predict.py accordingly to differ from model.py and added model details function into predict.py which could be called from pages.json and resolved the issue in same.

**Comment by abhisek2004** [comment]
Hi admin @yashasvini121 give me access i want to contribute to this project i will add Model Details in every page and other details. Can i add my own heart failure predction in this project . please give me acess......

**Comment by murtaza-sadri-19** [comment]
@yashasvini121, please review the changes

**Comment by murtaza-sadri-19** [comment]
@yashasvini121, I am working on changes and would commit it soon.

File: models/Breast_Cancer_Detector/predict.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    worst_perimeter = float(input("Enter worst perimeter: "))
+    worst_area = float(input("Enter worst area: "))
+    worst_smoothness = float(input("Enter worst smoothness: "))
+    worst_compactness = float(input("Enter worst compactness: "))
+    worst_concavity = float(input("Enter worst concavity: "))
+    worst_concave_points = float(input("Enter worst concave points: "))
+    worst_symmetry = float(input("Enter worst symmetry: "))
+    worst_fractal_dimension = float(input("Enter worst fractal dimension: "))
+
+    input_data = [mean_radius, mean_texture, mean_perimeter, mean_area, mean_smoothness, mean_compactness, mean_concavity, mean_concave_points, mean_symmetry, mean_fractal_dimension, radius_error, texture_error, perimeter_error, area_error, smoothness_error, compactness_error, concavity_error, concave_points_error, symmetry_error, fractal_dimension_error, worst_radius, worst_texture, worst_perimeter, worst_area, worst_smoothness, worst_compactness, worst_concavity, worst_concave_points, worst_symmetry, worst_fractal_dimension]
```
[the remark this answers is not in the document]
Comment (murtaza-sadri-19) [reply models/Breast_Cancer_Detector/predict.py:36]: The actual parameter which contribute to Cancer Detector are less so I am passing only required features to model.

**Comment by murtaza-sadri-19** [comment]
@yashasvini121, Changes made in PR #129

**Comment by abhisek2004** [comment]
@yashasvini121  Admin Can I add Heart Failure Predictive Project

**Comment by murtaza-sadri-19** [comment]
> @yashasvini121 Admin Can I add Heart Failure Predictive Project
@abhisek2004, you could open a new issue regarding that.

**Comment by abhisek2004** [comment]
ok

**Pull request closed without merging**